src/value.c

```diff
@@ -202,34 +202,36 @@ SIValue SIValue_Divide(const SIValue a, const SIValue b) {
 }
 
 int SIValue_Compare(const SIValue a, const SIValue b) {
-  /* In order to be comparable, both SIValues must be strings,
-   * booleans, or numerics. */
-  if (a.type == b.type) {
-    switch (a.type) {
-      case T_INT64:
-      case T_BOOL:
-        return a.longval - b.longval;
-      case T_DOUBLE:
-        return SAFE_COMPARISON_RESULT(a.doubleval - b.doubleval);
-      case T_STRING:
-        return strcmp(a.stringval, b.stringval);
-      case T_NODE:
-      case T_EDGE:
-        return ENTITY_GET_ID((GraphEntity*)a.ptrval) - ENTITY_GET_ID((GraphEntity*)b.ptrval);
-      default:
-        // Both inputs were of an incomparable type, like a pointer or NULL
-        return DISJOINT;
+    /* In order to be comparable, both SIValues must be strings,
+     * booleans, or numerics. */
+    if(a.type == b.type) {
+        switch (a.type) {
+        case T_INT64:
+        case T_BOOL:
+            return a.longval - b.longval;
+        case T_DOUBLE:
+            return SAFE_COMPARISON_RESULT(a.doubleval - b.doubleval);
+        case T_STRING:
+            return strcmp(a.stringval, b.stringval);
+        case T_NODE:
+        case T_EDGE:
+            return ENTITY_GET_ID((GraphEntity*)a.ptrval) - ENTITY_GET_ID((GraphEntity*)b.ptrval);
+        case T_NULL:
+            return 0;
```

Comment:
`NULL == NULL` is false in Cypher - https://neo4j.com/docs/cypher-manual/current/syntax/working-with-null/index.html#cypher-null-intro .

I think the caller should be responsible for managing DISJOINT returns.

Reply:
👍